Add Image Process

zhaohuxing opened the pull request.

pengsrc commented on helpers/image.go:
```diff
@@ -0,0 +1,265 @@
+	"reflect"
+	"strings"
+)
+
```
Import order is not tidy.

``` Go
import (
	"fmt"
	"reflect"
	"strings"

	"github.com/yunify/qingstor-sdk-go/service"
)
```

pengsrc commented on helpers/image_test.go:
```diff
@@ -0,0 +1,61 @@
+	qs "github.com/yunify/qingstor-sdk-go/service"
+	"testing"
+)
+
```
Import order is not tidy.

pengsrc commented on test/main.go:
```diff
@@ -58,7 +59,6 @@ func setUp() {
 	loadTestConfig()
 	loadConfig()
 	initQingStorService()
-
```
Why?

zhaohuxing committed:
Update the service codes

Add image process api codes generated by snips.
Update upload api codes generated by snips.

zhaohuxing committed:
Add image to the params to sign

Xuanwo commented on client/image.go:
```diff
@@ -0,0 +1,266 @@
+	ImageOutput *service.ImageProcessOutput
+	bucket      *service.Bucket
+	uri         *string
+	name        *string
```
Use `objectKey` directly here.

zhaohuxing replied:
OK, Thanks.

Xuanwo commented:
Rebase your code from master.

zhaohuxing committed:
Update git submodules

zhaohuxing committed:
Add usage docs of image process

zhaohuxing committed:
Add function of image process and testing
